Fix Trivy image scan workflow: update deprecated GitHub Actions

## Summary

The scheduled Trivy Image Scan workflow is using deprecated action versions, causing errors/warnings during CI runs. See: https://github.com/Apicurio/apicurio-registry/actions/runs/22940414589

## Problem

The reusable workflow at [`Apicurio/apicurio-gh-workflows/.github/workflows/image-scan.yaml`](https://github.com/Apicurio/apicurio-gh-workflows/blob/main/.github/workflows/image-scan.yaml) uses outdated action versions:

1. **`github/codeql-action/upload-sarif@v2`** — CodeQL Action v1 and v2 have been deprecated ([announcement](https://github.blog/changelog/2025-01-10-code-scanning-codeql-action-v2-is-now-deprecated/)). Must be updated to **`@v3`**.
2. **`aquasecurity/trivy-action@0.6.2`** — Very outdated version of the Trivy action. Should be updated to the latest release.
3. **`actions/checkout@v3`** — Should be updated to **`@v4`**.

## Error from CI

```
##[error]CodeQL Action major versions v1 and v2 have been deprecated. Please update all occurrences of the CodeQL Action in your workflow files to v3.
```

## Proposed Fix

Update the reusable workflow in the **[apicurio-gh-workflows](https://github.com/Apicurio/apicurio-gh-workflows)** repository (`/.github/workflows/image-scan.yaml`) to use current action versions:

- `actions/checkout@v3` → `actions/checkout@v4`
- `aquasecurity/trivy-action@0.6.2` → latest version
- `github/codeql-action/upload-sarif@v2` → `github/codeql-action/upload-sarif@v3`

**Note:** The fix is in the `apicurio-gh-workflows` repository, not in this repository. The workflow file in this repo (`.github/workflows/image-scan.yaml`) simply calls the reusable workflow.

--- a/.github/workflows/image-scan.yaml
+++ b/.github/workflows/image-scan.yaml
@@ -6,7 +6,24 @@ on:
     - cron: "0 6 * * *"
 
 jobs:
-  call-image-scan-registry:
-    uses: Apicurio/apicurio-gh-workflows/.github/workflows/image-scan.yaml@main
-    with:
-      image: quay.io/apicurio/apicurio-registry:latest-snapshot
+  vulnerability-scan:
+    name: Container vulnerability scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout "${{ github.ref }}"
+        uses: actions/checkout@v4
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.29.0
+        with:
+          image-ref: quay.io/apicurio/apicurio-registry:latest-snapshot
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          severity: 'CRITICAL,HIGH'
+          vuln-type: 'os,library'
+          ignore-unfixed: true
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v3
+        with:
+          sarif_file: 'trivy-results.sarif'

